### PR TITLE
Correct role name

### PR DESCRIPTION
**Pull request by weiguoz**

### CODE REVIEW

File: go/codegen/pai/tensorflow.go [hunk 1]
```diff
@@ -236,7 +236,20 @@ func getCheckpointDir(ossModelPath, project string) (string, error) {
...
-	roleName := fmt.Sprintf("pai2oss_%s", project)
+	roleName := genRoleName(project)
 	// format the oss checkpoint path with ARN authorization.
 	return fmt.Sprintf("%s/?role_arn=%s/%s&host=%s", ossURL, ra.Arn, roleName, ra.Host), nil
 }
+
+func genRoleName(project string) string {
+	var rn bytes.Buffer
+	rn.WriteString("pai2oss")
+	for _, ch := range project {
```
Comment (shendiaomo) [inline go/codegen/pai/tensorflow.go:249]: How about using `regexp` to replace this loop like:
```Go
re := regexp.MustCompile(`[^a-z0-9]`)
rn.WriteString(re.ReplaceAllString(strings.ToLower(project), ""))
```

File: go/codegen/pai/tensorflow_test.go [hunk 2]
```diff
@@ -29,6 +29,15 @@ func TestGetCheckpointDir(t *testing.T) {
...
 	a.Equal(expectedCkp, ckpoint)
 }
+
+func TestGenRoleName(t *testing.T) {
+	a := assert.New(t)
+	a.Equal(genRoleName("a"), "pai2ossa")
+	a.Equal(genRoleName("abc"), "pai2ossabc")
+	a.Equal(genRoleName("a_b_c"), "pai2ossabc")
+	a.Equal(genRoleName("a_b_C"), "pai2ossabc")
+	a.Equal(genRoleName("a+b2_C"), "pai2ossab2c")
```
Comment (shendiaomo) [inline go/codegen/pai/tensorflow_test.go:42]: Are these automatic conversions desired? What if we just report an error when characters except `[a-z0-9]` are present.

File: go/codegen/pai/tensorflow.go (hunk 1 above)
Comment (weiguoz) [reply go/codegen/pai/tensorflow.go:249]: It is a concise way, especially one pass, and easy to understand.
So I didn't choose using `regexp`.

File: go/codegen/pai/tensorflow_test.go (hunk 2 above)
Comment (weiguoz) [reply go/codegen/pai/tensorflow_test.go:42]: The `project` name contains `_` often. We should remove such characters to generate a valid role name.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**